csi.proto

```diff
@@ -635,6 +671,25 @@ message NodeUnpublishVolumeRequest {
   string target_path = 2;
 }
 
+message NodeExpandVolumeRequest {
+  // The ID of the volume. This field is REQUIRED
+  string volume_id = 1;
+
+  // the current capacity of the volume in bytes.
+  // This field is OPTIONAL
+  int64 current_capacity_bytes = 2;
+
+  // This field is OPTIONAL. This allows CO to specify the
+  // capacity requirements of the volume after expansion
+  CapacityRange capacity_range = 3;
```

Comment:
I'm not convinced that this is needed, really. The proposed controller API already provides this field.

Perhaps, instead, the ControllerExpandVolumeResponse could return an opaque "expansion_parameters" field that would be specified as part of this request, for use with any SP-specific tooling that required proprietary metadata in order to carry out follow-up operations on the node (like a filesystem resize)?

Reply:
I added field with the expectation that - some file systems may require current and expected capacity for expanding the file system. Already for example `xfs_growfs` invocation looks like:

```
~> xfs_growfs /mount/point -D size
```

If we omit size from the argument then it is expanded to the end of volume. In many cases, I do agree that file system expansion in CSI driver could work without expected capacity but there can be cases where it needs the expected capacity too. 

Also, if we were to rename this feature to `resize` from `expand` it will be absolutely needed. 

I am not sure how `ControllerExpandVolumeResponse` will be accessible to node driver. I am assuming that most of CSI spec is designed to be stateless.